cassandra/cluster.py

```diff
@@ -3993,6 +4007,34 @@ def _refresh_node_list_and_token_map(self, connection, preloaded_results=None,
             log.debug("[control connection] Rebuilding token map due to topology changes")
             self._cluster.metadata.rebuild_token_map(partitioner, token_map)
 
+    # Experimental, this interface and use may change
+    def _init_tablets(self):
+        self._cluster.metadata._tablets = Tablets({})
+
+    # Experimental, this interface and use may change
+    def add_tablet(self, keyspace, table, tablet):
```

Comment:
What is the concurrency situation of `_set_result`? Is it possible for it to be called from 2 threads simultaneously? If so, `Tablets` class (where this function should be) needs to handle it, and right now it doesn't.